addons/mrp_account/models/stock_move.py

```diff
@@ -44,3 +44,22 @@ def _is_production(self):
     def _is_production_consumed(self):
         self.ensure_one()
         return self.location_dest_id.usage == 'production' and self.location_id._should_be_valued()
+
+    def _get_out_svl_vals(self, forced_quantity):
+        unbuild_moves = self.filtered('unbuild_id')
+        # `price_unit` of finished product moves @ build time
+        prod_price_unit_map = {
+            move.product_id.id: move.stock_valuation_layer_ids[0].unit_cost
+            for move in unbuild_moves.unbuild_id.mo_id.move_finished_ids.sudo()
+            if move.product_id.categ_id.property_cost_method != 'standard' and
+            move.stock_valuation_layer_ids
+        }
```

Comment:
```suggestion
        prod_price_unit_map = {
            move.id: move.unbuild_id.mo_id.move_finished_ids.sudo() stock_valuation_layer_ids.filtered(lambda l: l.product_id == move.product_id)[0].unit_cost
            for move in unbuild_move
            if move.product_id.cost_method != 'standard' and
            move.unbuild_id.mo_id.move_finished_ids.stock_valuation_layer_ids and 
        }
```
in case you validate multiple unbuilds targeting the same finished product. I also add a `filtered` to avoid getting byproduct`  unit_cost